### PR TITLE
Create initial structure and make files

### PR DESCRIPTION
**Pull request by becauselol**
im tired of working on this branch
also want to give you guys something to work with
implemented
- paragraph
- headers (but there's a bug)
- code block (but there's a bug also)

### CODE REVIEW

File: src/parser/features/codeblock.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+#include <string.h>
+#include "codeblock.h"
+
+int is_codeblock_indicator(const char *line, int line_length) {
+  if (line_length < 3) {
+    return 0;
+  }
+  return strstr(line, "```") != NULL && line[0] == '`';
```
Comment (DeProfessorA) [inline src/parser/features/codeblock.c:9]: What if line = '`abc```'? Or is this just not allowed?

File: src/parser/features/headers.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  if (line == NULL) {
+    return 0;
+  }
+  if (line[0] != '#') {
+    return 0;
+  }
+  int have_hex = 1;
+  int counter = 0;
+  for (int i = 0; i < line_length; i++) {
+    if (have_hex && line[i] == '#') {
```
Comment (DeProfessorA) [inline src/parser/features/headers.c:28]:
```
nit: Simpler version?

 if (line[i] == '#') {
      counter++;
      continue;
    } else if (line[i] == ' ') {
      break;
    } else {
      have_hex = 0;
      break;
    }
```

File: src/parser/features/codeblock.c (hunk 1 above)
Comment (becauselol) [reply src/parser/features/codeblock.c:9]: just not allowed

File: src/parser/features/headers.c (hunk 2 above)
Comment (becauselol) [reply src/parser/features/headers.c:28]: ok will change

**Review by DeProfessorA — APPROVED** [review]
LGTM

**Pull request merged**